File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-2.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int len = s.length();
        if(len==0 || s==null) return 0;
        int left =0;
        int max=0;

        Set<Character> set = new HashSet<>();

        for(int right =0;right<len;right++){
            char ch = s.charAt(right);
            while(set.contains(ch)){
                set.remove(s.charAt(left));
                left++;
                
            }
            set.add(ch);
            max=Math.max(max,right-left+1);


        }
        
        return max;
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.kt
class Solution {
    fun longestConsecutive(nums: IntArray): Int {
      if(nums.size==0 && nums==null) return 0
      
      val set = HashSet<Int>()

      for(i in nums){
        set.add(i)
      }

      var best = 0

      for(j in nums){
        if(!set.contains(j-1)){
        var cur =j
        var size=1
        while(set.contains(cur+1)){
          cur++
          size++
        }

        best = Math.max(best,size)
      }
    }
    return best
}
}
